src/Database/QueryCompiler.php

```diff
@@ -270,6 +270,29 @@ protected function _buildJoinPart(array $parts, Query $query, ValueBinder $gener
         return $joins;
     }
 
+    /**
+     * Helper function to build the string representation of a window clause.
+     *
+     * @param array $parts List of windows to be transformed to string
+     * @param \Cake\Database\Query $query The query that is being compiled
+     * @param \Cake\Database\ValueBinder $generator the placeholder generator to be used in expressions
+     * @return string
+     */
+    protected function _buildWindowPart(array $parts, Query $query, ValueBinder $generator): string
+    {
+        $windows = [];
+        foreach ($parts as $window) {
+            $expr = $window['window'];
```

Comment:
Should this also have a runtime check to ensure we don't have a fatal error/type error on the method calls?

Reply:
If we add a type check in `Query::window()` we shouldn't need one here.